js/encoder.h

```diff
@@ -0,0 +1,152 @@
+#pragma once
+
+#include <nan.h>
+#include <cmath>
+#include "../cpp/encoder.h"
+
+using namespace v8;
+
+class Encoder {
+    static const size_t DEFAULT_RECURSE_LIMIT = 256;
+    static const size_t INITIAL_BUFFER_SIZE = 1024 * 1024;
+    static const size_t MAX_SIZE = (2L << 32) - 1;
+
+public:
+    Encoder() {
+        ret = 0;
+        pk.buf = (char*)malloc(INITIAL_BUFFER_SIZE);
```

Comment:
I meant, you have an init and a deinit here for the data inside of pk, but otherwise that data is all handled within those erlpack_buffer_* functions. I think it would make more sense for init and deinit to be in erlpack_buffer_* functions too.